frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/PhoneNumberCodeProcessor.java

```diff
@@ -99,29 +138,49 @@ public Optional<ErrorResponse> validateCode() {
 
     @Override
     public void processSuccessfulCodeRequest(String ipAddress, String persistentSessionId) {
-        switch (codeRequest.getJourneyType()) {
-            case REGISTRATION:
-                dynamoService.updatePhoneNumberAndAccountVerifiedStatus(
-                        emailAddress, codeRequest.getProfileInformation(), true, true);
-                submitAuditEvent(
-                        FrontendAuditableEvent.UPDATE_PROFILE_PHONE_NUMBER,
-                        MFAMethodType.SMS,
-                        codeRequest.getProfileInformation(),
-                        ipAddress,
-                        persistentSessionId,
-                        false);
-                break;
-            case ACCOUNT_RECOVERY:
-                dynamoService.setVerifiedPhoneNumberAndRemoveAuthAppIfPresent(
-                        emailAddress, codeRequest.getProfileInformation());
-                submitAuditEvent(
-                        FrontendAuditableEvent.UPDATE_PROFILE_PHONE_NUMBER,
-                        MFAMethodType.SMS,
-                        codeRequest.getProfileInformation(),
-                        ipAddress,
-                        persistentSessionId,
-                        true);
-                break;
+        if (codeRequest.getJourneyType() == JourneyType.REGISTRATION
+                || codeRequest.getJourneyType() == JourneyType.ACCOUNT_RECOVERY) {
+            JourneyType journeyType = codeRequest.getJourneyType();
+            UserProfile userProfile = userContext.getUserProfile().get();
+            boolean phoneNumberVerified = userProfile.isPhoneNumberVerified();
+            String phoneNumber =
+                    PhoneNumberHelper.formatPhoneNumber(codeRequest.getProfileInformation());
+            boolean updatedPhoneNumber = !phoneNumber.equals(userProfile.getPhoneNumber());
+
+            if (configurationService.isPhoneCheckerWithReplyEnabled()
```

Comment:
I wonder particularly about pulling this check of the feature flag into the top `if` - I think it might be a bit easier to read if you could immediately see that this whole block from line 141 was only relevant with the feature flag on.

Reply:
This is the scenario I was talking about in standup. There could be a risk of regression then as currently you can update your current phone number with the same number during account recovery. Ideally we need a small story to validate this on the ui (or maybe a new errorcode on api) so that the user receives the correct feedback.
I have pulled out the sqs code into another method for visibility though.

Reply:
Ah got you, yeah I was specifically thinking just that the feature flag check could go to the top but I think I was misunderstanding the scope of the overall if :-) The version with the method pulled out looks good, nice and readable